llvm/test/tools/sycl-post-link/device-globals/test_global_variable_many_kernels_in_one_module.ll

```diff
@@ -0,0 +1,189 @@
+; RUN: sycl-post-link --device-globals --split=source -S %s -o %t.files.table
+; RUN: FileCheck %s -input-file=%t.files_0.ll --check-prefix CHECK-MOD0
+; RUN: FileCheck %s -input-file=%t.files_1.ll --check-prefix CHECK-MOD1
+
+; This test is intended to check that sycl-post-link generates no errors
+; when a device global variable with the 'device_image_scope' property
+; is used in many kernels grouped to one module.
+
+target datalayout = "e-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024-n8:16:32:64"
+target triple = "spir64-unknown-unknown"
+
+%"class.cl::sycl::ext::oneapi::device_global" = type { i32 }
+%"class.cl::sycl::range" = type { %"class.cl::sycl::detail::array" }
+%"class.cl::sycl::detail::array" = type { [1 x i64] }
+%"class.cl::sycl::detail::accessor_common" = type { i8 }
+
+$_ZN2cl4sycl2idILi1EEC2ILi1EEENSt9enable_ifIXeqT_Li1EEyE4typeE = comdat any
+$_ZN2cl4sycl6detail5arrayILi1EEC2ILi1EEENSt9enable_ifIXeqT_Li1EEyE4typeE = comdat any
+$_ZN2cl4sycl2idILi1EEC2Ev = comdat any
+$_ZN2cl4sycl6detail14InitializedValILi1ENS0_5rangeEE3getILi0EEENS3_ILi1EEEv = comdat any
+$_ZN2cl4sycl5rangeILi1EEC2ILi1EEENSt9enable_ifIXeqT_Li1EEyE4typeE = comdat any
+$_ZTSZ7kernel1RN2cl4sycl5queueEEUlvE_ = comdat any
+$_ZTSZ7kernel2RN2cl4sycl5queueEEUlvE_ = comdat any
+$_ZTSZ7kernel3RN2cl4sycl5queueEEUlvE_ = comdat any
+$_ZTSZ7kernel4RN2cl4sycl5queueEEUlvE_ = comdat any
+
+$dg_int2 = comdat any
+@dg_int2 = linkonce_odr dso_local addrspace(1) constant %"class.cl::sycl::ext::oneapi::device_global" zeroinitializer, comdat, align 4 #0
+; CHECK-MOD0: @dg_int2 = linkonce_odr dso_local addrspace(1) constant %"class.cl::sycl::ext::oneapi::device_global" zeroinitializer, comdat, align 4
+; CHECK-MOD1-NOT: @dg_int2
+
+; Third kernel that uses no device-global variables
+define weak_odr dso_local spir_kernel void @_ZTSZ7kernel3RN2cl4sycl5queueEEUlvE_() #6 comdat !kernel_arg_buffer_location !6 {
+entry:
+  ret void
+}
+
+; Function is required to increase the function list only
```

Comment:
Now, I'm not sure. I increased the function list because I planned to implement a naive algorithm to find all the usages of device global variables but LLVM can do it for me by "def-use" chains. When the module graph is traversed, any functions without references to device global variables just are ignored.

Thank you, I've removed those "extra" function from the lit tests.